### PR TITLE
Server auth example

### PR DESCRIPTION
**Pull request by colfax23**

### CODE REVIEW

File: pages/api/authenticate.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    sessionId: string,
+    userId: string
+  }
+};
+
+export type AuthenticationErrorResponse = {
+  error: string;
+};
+
+export default async function GET(req: NextApiRequest, res: NextApiResponse<AuthenticateSuccessResponse | AuthenticationErrorResponse>) {
```
Comment (JoshNavi) [inline pages/api/authenticate.ts:24]: nit, I would rename this `handler` the `GET`/`POST` etc. naming is an app router paradigm and this endpoint will get called for any http method at that path. Nothing wrong about this example just not very idiomatic

**Review by JoshNavi — APPROVED** [review]
another nit is that I wouldnt call the endpoint `authenticate` since like it doesnt do any authentication it just can read authenticated state, but thats not so important

**Review by Koleok — APPROVED** [review]
Would be good to also show a cookie version in a `getServerSideProps` to handle redirects etc but this does the job ❤️

File: pages/dashboard.tsx [hunk 2]
```diff
@@ -41,6 +41,24 @@ export default function DashboardPage() {
   const twitterSubject = user?.twitter?.subject || null;
   const discordSubject = user?.discord?.subject || null;
 
+  const authenticateWithServer = async () => {
```
Comment (ankushswar1) [inline pages/dashboard.tsx:44]: nit: I would rename this `authorizeWithServer` just for consistency with how we frame it in docs

File: pages/index.tsx [hunk 3]
```diff
@@ -1,9 +1,30 @@
...
+  // Use this result to pass props to a page for server rendering or to drive redirects!
+  // ref https://nextjs.org/docs/pages/api-reference/functions/get-server-side-props
+  const claims = await client.verifyAuthToken(cookieAuthToken);
+  console.log({ claims });
+
+  return {
+    props: {},
+    redirect: { destination: "/dashboard", permanent: false },
+  };
+};
```
Comment (JoshNavi) [inline pages/index.tsx:24]: pages API was so much better at so many things when it make to stuff like this, nextjs middleware makes me sad

**Review by JoshNavi — APPROVED** [review]

File: pages/index.tsx (hunk 3 above)
Comment (JoshNavi) [reply pages/index.tsx:24]: tbh makes me wanna go back to developing actual servers using actual server frameworks

File: pages/index.tsx [hunk 4]
```diff
@@ -1,9 +1,30 @@
...
+  const cookieAuthToken = req.cookies["privy-token"];
+  if (!cookieAuthToken) return { props: {} };
+
+  const PRIVY_APP_ID = process.env.NEXT_PUBLIC_PRIVY_APP_ID;
+  const PRIVY_APP_SECRET = process.env.PRIVY_APP_SECRET;
+  const client = new PrivyClient(PRIVY_APP_ID!, PRIVY_APP_SECRET!);
+
+  // Use this result to pass props to a page for server rendering or to drive redirects!
+  // ref https://nextjs.org/docs/pages/api-reference/functions/get-server-side-props
+  const claims = await client.verifyAuthToken(cookieAuthToken);
```
Comment (ankushswar1) [inline pages/index.tsx:17]: supernit: should we `try`/`catch` this and then route the user depending on if error or success

**Review by ankushswar1 — APPROVED** [review]

File: pages/index.tsx (hunk 3 above)
Comment (Koleok) [reply pages/index.tsx:24]: You know I really agree, this is so easy to understand

**Pull request merged**